File: code/widgets/org.eclipsescout.demo.widgets.client/src/org/eclipsescout/demo/widgets/client/ui/forms/ListBoxForm.java
/*******************************************************************************
 * Copyright (c) 2015 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Distribution License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/org/documents/edl-v10.html
 * 
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 ******************************************************************************/
package org.eclipsescout.demo.widgets.client.ui.forms;

import org.eclipse.scout.commons.annotations.Order;
import org.eclipse.scout.commons.exception.ProcessingException;
import org.eclipse.scout.rt.client.ui.form.AbstractForm;
import org.eclipse.scout.rt.client.ui.form.AbstractFormHandler;
import org.eclipse.scout.rt.client.ui.form.fields.button.AbstractCloseButton;
import org.eclipse.scout.rt.client.ui.form.fields.groupbox.AbstractGroupBox;
import org.eclipse.scout.rt.client.ui.form.fields.listbox.AbstractListBox;
import org.eclipse.scout.rt.shared.TEXTS;
import org.eclipse.scout.rt.shared.services.lookup.LookupCall;
import org.eclipsescout.demo.widgets.client.services.lookup.CompanyTypeLookupCall;
import org.eclipsescout.demo.widgets.client.ui.forms.ListBoxForm.MainBox.CloseButton;
import org.eclipsescout.demo.widgets.client.ui.forms.ListBoxForm.MainBox.GroupBox;
import org.eclipsescout.demo.widgets.client.ui.forms.ListBoxForm.MainBox.GroupBox.ListBoxField;

public class ListBoxForm extends AbstractForm implements IPageForm {

  public ListBoxForm() throws ProcessingException {
    super();
  }

  @Override
  protected boolean getConfiguredAskIfNeedSave() {
    return false;
  }

  @Override
  protected String getConfiguredTitle() {
    return TEXTS.get("ListBox");
  }

  @Override
  public void startPageForm() throws ProcessingException {
    startInternal(new PageFormHandler());
  }

  @Override
  public CloseButton getCloseButton() {
    return getFieldByClass(CloseButton.class);
  }

  public GroupBox getGroupBox() {
    return getFieldByClass(GroupBox.class);
  }

  public ListBoxField getListBoxField() {
    return getFieldByClass(ListBoxField.class);
  }

  public MainBox getMainBox() {
    return getFieldByClass(MainBox.class);
  }

  @Order(10.0)
  public class MainBox extends AbstractGroupBox {

    @Order(10.0)
    public class GroupBox extends AbstractGroupBox {

      @Order(10.0)
      public class ListBoxField extends AbstractListBox<Long> {

        @Override
        protected int getConfiguredGridH() {
          return 5;
        }

        @Override
        protected int getConfiguredGridW() {
          return 2;
        }

        @Override
        protected String getConfiguredLabel() {
          return TEXTS.get("ListBox");
        }

        @Override
        protected Class<? extends LookupCall> getConfiguredLookupCall() {
          return CompanyTypeLookupCall.class;
        }
      }
    }

    @Order(20.0)
    public class CloseButton extends AbstractCloseButton {
    }
  }

  public class PageFormHandler extends AbstractFormHandler {
  }
}

File: modules/common/examples/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/listbox/ListBoxForm.js
/*
 * Copyright (c) 2017 BSI Business Systems Integration AG.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Distribution License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/org/documents/edl-v10.html
 *
 * Contributors:
 *     BSI Business Systems Integration AG - initial API and implementation
 */
import {Form, models} from '@eclipse-scout/core';
import ListBoxFormModel from './ListBoxFormModel';

export default class ListBoxForm extends Form {

  constructor() {
    super();
  }

  _jsonModel() {
    return models.get(ListBoxFormModel);
  }

  // noinspection DuplicatedCode
  _init(model) {
    super._init(model);

    this.listBox = this.widget('ListBox');

    this.lookupCallField = this.widget('LookupCallField');
    this.lookupCallField.setValue(this.listBox.lookupCall);
    this.lookupCallField.on('propertyChange:value', event => this.listBox.setLookupCall(event.newValue));
    this.listBox.on('propertyChange:lookupCall', event => this.lookupCallField.setValue(event.newValue));

    this.widget('ValueFieldPropertiesBox').setField(this.listBox);
    this.widget('FormFieldPropertiesBox').setField(this.listBox);
    this.widget('PropertiesBox').setTable(this.listBox.table);
    this.widget('GridDataBox').setField(this.listBox);
    this.widget('WidgetActionsBox').setField(this.listBox);
    this.widget('FormFieldActionsBox').setField(this.listBox);
    this.widget('EventsTab').setField(this.listBox);
  }
}
